backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/projection/impl/LuceneIdProjection.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.search.projection.impl;
+
+import org.hibernate.search.backend.lucene.lowlevel.collector.impl.DocumentReferenceCollector;
+import org.hibernate.search.backend.lucene.search.extraction.impl.LuceneResult;
+import org.hibernate.search.backend.lucene.search.impl.LuceneSearchContext;
+import org.hibernate.search.engine.backend.common.DocumentReference;
+import org.hibernate.search.engine.search.loading.spi.LoadingResult;
+import org.hibernate.search.engine.search.loading.spi.ProjectionHitMapper;
+import org.hibernate.search.engine.search.projection.SearchProjection;
+import org.hibernate.search.engine.search.projection.spi.IdProjectionBuilder;
+
+public class LuceneIdProjection<I> extends AbstractLuceneProjection<DocumentReference, I> {
+
+	private final Class<I> identifierType;
+
+	private LuceneIdProjection(LuceneSearchContext searchContext, Class<I> identifierType) {
+		super( searchContext );
+		this.identifierType = identifierType;
+	}
+
+	@Override
+	public String toString() {
+		return getClass().getSimpleName();
+	}
+
+	@Override
+	public void request(SearchProjectionRequestContext context) {
+		context.requireCollector( DocumentReferenceCollector.FACTORY );
```

Comment:
Similarly to my comment on the Elasticsearch backend, I would use a different collector here: one that does not collect document references, but document IDs only (Strings). You could copy/paste `DocumentReferenceCollector` and remove everything related to the entity type (`metadataResolver`, `currentLeafMappedTypeName`, ...).

Reply:
> I'm not sure that the document ids are enough to load the entity ids. I think that we still need DocumentReferences.

Reply:
Same answer.